### PR TITLE
feat(sdk): Add pipeline_task_name to PipelineTaskFinalStatus

### PR DESCRIPTION
**Pull request by chensun**
**Description of your changes:**

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: sdk/python/requirements.in [hunk 1]
```diff
@@ -16,7 +16,7 @@ google-cloud-storage>=1.20.0,<2
 # NOTE: Maintainers, please do not require google-auth>=2.x.x
 # Until this issue is closed
 # https://github.com/googleapis/google-cloud-python/issues/10566
-kfp-pipeline-spec>=0.1.13,<0.2.0
+kfp-pipeline-spec>=0.1.14,<0.2.0
```
Comment (chensun) [inline sdk/python/requirements.in:19]: This change itself doesn't have a hard dependency on the new version, but it's related.

**Review by connor-mccarthy — APPROVED** [review]
/lgtm

**Comment by connor-mccarthy** [comment]
@chensun, looks good! Approve when you'd like to merge.

/remove-approve
/lgtm
/unhold

**Comment by ji-yaqi** [comment]
/lgtm
/approve

Thanks!

**Pull request merged**